pkg/logql/log/filter.go

```diff
@@ -335,14 +335,32 @@ func parseRegexpFilter(re string, match bool) (Filterer, error) {
 	// attempt to improve regex with tricks
 	f, ok := simplify(reg)
 	if !ok {
-		return newRegexpFilter(re, match)
+		allNonGreedy(reg)
+		return newRegexpFilter(reg.String(), match)
 	}
 	if match {
 		return f, nil
 	}
 	return newNotFilter(f), nil
 }
 
+// allNonGreedy turns greedy qunatifiers such as `.*` and `.+` into non-greedy ones. This is the same effect as writing
+// `.*?` and `.+?`. This is only save because we use `Match`. If we were to find the exact position and lenght of the match
```

Comment:
```suggestion
// `.*?` and `.+?`. This is only safe because we use `Match`. If we were to find the exact position and length of the match
```

Reply:
I think you should include the justification, that `Match` can return quicker if we don't require the longest match.
(IMO this is a quirk or implementation detail of the Go standard library).